processing/src/main/java/io/druid/segment/incremental/OnheapIncrementalIndex.java

```diff
@@ -238,10 +303,26 @@ protected void concurrentRemove(int offset)
   @Override
   public boolean canAppendRow()
   {
-    final boolean canAdd = size() < maxRowCount;
-    if (!canAdd) {
-      outOfRowsReason = StringUtils.format("Maximum number of rows [%d] reached", maxRowCount);
+    final boolean countCheck = size() < maxRowCount;
+    boolean sizeCheck = true;
```

Comment:
This can be simplified into `final boolean sizeCheck = maxBytesInMemory > 0 && getBytesInMemory() < maxBytesInMemory`;

Reply:
`final boolean sizeCheck = maxBytesInMemory > 0 && getBytesInMemory() < maxBytesInMemory;` changes the logic, because if `maxBytesInMemory = -1 `, we want to ignore sizeCheck and not have `canAppendRow()` return false.

Reply:
Ah, then `final boolean sizeCheck = maxBytesInMemory <= 0 || getBytesInMemory() < maxBytesInMemory`?